geopandas/tests/test_pandas_methods.py

```diff
@@ -603,6 +603,11 @@ def test_apply_preserves_geom_col_name(df):
     assert result.geometry.name == "geom"
 
 
+def test_df_apply_returning_series(df):
+    # https://github.com/geopandas/geopandas/issues/2283
+    df.apply(lambda row: row.geometry, axis=1)
```

Comment:
```suggestion
    result = df.apply(lambda row: row.geometry, axis=1)
    assert_geoseries_equal(result, df.geometry, check_crs=False)
```
I think it is better to check the actual result, rather than checking if it raises or not.